File: CPP [1501-2000]/1957.delete_characters_to_make_fancy_string [2].cpp
class Solution {
public:
    // T.C.=O(n), S.C.=O(1)
    string makeFancyString(string s) {
        char prev2 = '#';
        char prev1 = '#';
        string ans = "";

        for(char ch: s)
        {
            if(prev1 == '#')
            {
                prev1 = ch;
                ans += ch;
            }
            else if(prev2 == '#' || prev1 != ch || prev2 != ch)
            {
                prev2 = prev1;
                prev1 = ch;
                ans += ch;
            }
        }

        return ans;
    }
};

File: JS/1957.delete_characters_to_make_fancy_string [2].js
/**
 * @param {string} s
 * @return {string}
 */
var makeFancyString = function(s) {
        let prev2 = '#'
        let prev1 = '#'
        let ans = ''

        for(const ch of s){
            if(prev1 === '#'){
                prev1 = ch
                ans += ch
            }
            else if(prev2 === '#' || prev1 !== ch || prev2 !== ch){
                prev2 = prev1
                prev1 = ch
                ans += ch
            }
        }

        return ans
};